scripts/internal/prune_size_model.py

```diff
@@ -0,0 +1,246 @@
+#!/usr/bin/env python
+
+from __future__ import print_function
+import math, sys
+
+
+"""
+This builds a model predicting the num-ngrams that we will get from doing a
```

Comment:
does it *build* a model?  I thought in what we were discussing, it was a model fixed in advance, that we only adjust slightly in case of overshoot.
I'm concerned that if we estimate the model on the fly, there are too many possibilities for instability.  But I'm OK to try it.